dips/dip-1.mdx

```diff
@@ -211,18 +211,88 @@ Represents an error that occurred in response to a Command.
+
+`invalid_x_request_sender_address`:
+* `X-REQUEST-SENDER-ADDRESS` header value is not the request sender’s address in the command object. All command objects should have a field that is the request sender’s address. For payment object, it is `sender#address` or `receiver#address`.
+* Could not find Diem's onchain account by the `X-REQUEST-SENDER-ADDRESS` header value.
+* Could not find the compliance key of the onchain account found by the `X-REQUEST-SENDER-ADDRESS` header value.
+* The compliance key found from the onchain account is not a valid ED25519 public key.
+
+`missing_http_header`: missing HTTP header `X-REQUEST-ID` or `X-REQUEST-SENDER-ADDRESS`.
+
+##### JWS Validation Error Codes
+
+`invalid_jws`: invalid JWS format (compact) or protected header
+
+`invalid_jws_signature`: JWS signature verification failed
+
+##### Request Object Validation Error Codes
+
+`invalid_json`: decoded JWS body is not json.
+
+`invalid_object`: command request/response object json is not object, or the command object type does not match `command_type`.
+
+`missing_field`:
+* Missing required field.
+* An optional field is required to be set for a specific state, e.g. PaymentObject requires sender's `kyc_data` (which is an optional field for `PaymentActorObject`) when sender init the `PaymentObject`.
+
+`unknown_field`: field is unknown for an object.
+
+`unknown_command_type`: invalid/unsupported `command_type`.
+
+`invalid_field_value`:
+* Invalid / unknown enum field values.
+* UUID field value does not match UUID format.
+* Payment actor address is not a valid DIP-5 account identifier.
+* Currency field value is not a valid Diem currency code for the connected network.
+
+`invalid_command_producer`: The HTTP request sender is not the right actor to send the payment object. For example, if the actor receiver sends a new command with payment object change that should be done by actor sender.
+
+`invalid_initial_or_prior_not_found`: could not find command by reference_id for a non-initial state command object; for example, actor receiver received a payment command object that actor sender status is `ready_for_settlement`, but receiver could not find any command object by the reference id.
+
+`no_kyc_needed`: payment action amount is under travel rule limit.
+
+`invalid_recipient_signature`:
+* Field `recipient_signature` value is not hex-encoded bytes.
+* Field `recipient_signature` value is an invalid signature.
+
+`unknown_address`:
+* The DIP-5 account identifier address in the command object is not HTTP request sender’s address or receiver’s address. For payment object it is `sender#address` or `receiver#address`.
+* Could not find on-chain account by an DIP-5 account identifier address in command object address.
+
+`conflict`:
+* Command object is in conflict with another different command object by cid, likely a cid is reused for different command object.
+* Failed to acquire lock by for the command object by the `reference_id`.
```

Comment:
This failed to acquire lock confuses me: now we do not have sender / receiver , client / server competing for objects? So when can a lock fail?

Reply:
we should remove this

Reply:
For example: 
* VASP A sends sinit to VASP B
* VASP B side has a bug, it sends 2 commands at almost same time:
  *  rsend
  *  rsoft
*  VASP B received the command requests, if it does not lock the previous sinit before start to validate and save the command, then 2 requests are processed, one of them overwrites another one, the result state is unknown.
* VASP B received the success responses, and save the result, one of them will be saved.
* It is possible we ende with one of the following result: 
  *  VASP A: rsend, VASP B: rsoft
  * VASP A: rsoft, VASP B: rsend
* Once we enter the above inconsistent result, VASP A's following up command request will always fail.